contracts/Core/RewardManager.sol

```diff
@@ -86,6 +88,7 @@ contract RewardManager is Initializable, Constants, RewardManagerParams, IReward
         // uint256 currentStake = previousStake;
         if (newStake < previousStake) {
             stakeManager.setStakerEpochFirstStakedOrLastPenalized(epoch, stakerId);
+            if (randao) stakeManager.setStakerStake(epoch, stakerId, StakeChanged.RandaoPenalty, previousStake, newStake);
```

Comment:
Hey this is not ideal 
SetStatkerStake is going to be called twice for setting same newStake

Reply:
Fixed @abhishekvispute

Reply:
reason = randao ? StakeChanged.RandaoPenalty : StakeChanged.InactivityPenalty

stakeManager.setStakerStake(epoch, stakerId, StakeChanged.InactivityPenalty, previousStake, newStake);

Reply:
Hey here reason cannot be of string type as explicit conversion is needed and it shows enum cannot be converted to string.
I tried to include in enum too it's not working (type error)

Reply:
Hey we can't have setStakerStake outside the if block as it will be called unnecessarily even if there is no penalty and thus each time a false event will be emitted if no penalty is there.